### ISSUE TITLE
Mock browser environment on build

### ISSUE THREAD

**Issue by iansinnott** [iansinnott/react-static-webpack-plugin#31]
**TL;DR:** Any code that requires a browser environment will break the render and it's not often clear where the breakage occurred. This is especially true for third-party modules. I propose we render all code within a mocked browser environment using JSDOM.

**Primary concerns to address:**

* Third party libraries that depend on the browser environment
* Opaque error propagation does not make it clear what package is breaking the render
* Requiring the user to worry about whether they are rending in node or in the browser goes against the mission statement of this project

**Full Description**

This plugin currently requires you to think about your app as if it were being rendered on the server. Meaning you have to occasionally write code to _check_ whether or not you are running code in a node process or in the browser. Example:

```js
if (typeof document !== 'undefined') {
  ReactDOM.render(<App />, document.querySelector('#root'));
}
```

This has generally been fine, since it's not too cumbersome to write occasional checks like this. You can also put DOM interaction code within `componentDidMount` and it will run just fine since `componentDidMount` is not called during static rendering. 

However, a more serious problem arrises when third party libraries depend on a browser environment. As an example, I recently added `auth0-js` to a static project to handle auth, but it does _not_ support server side rendering at all so I had to wrap all of my use of the code in a custom mock to get it to render correctly.

So, I'm proposing rendering all webpacked code in a mocked browser environment using JSDOM to allow almost _any_ code that works in a browser to render through this plugin. Of course actual DOM interaction should still be done once components are mounted. 

Technically speaking this could be achieved by augmenting the node VM [`sandbox`](https://github.com/iansinnott/react-static-webpack-plugin/blob/master/src/utils.js#L156) with JSDOM. Something along the lines of:

```js
const document = require('jsdom').jsdom('<body></body>');

sandbox.document = document;
sandbox.window = document.defaultView;
sandbox.navigator = window.navigator;
```

**Comment by iansinnott** [comment]
@BerkeleyTrue what do you think? You have more experience rendering components within Node
